sdk/messaging/azservicebus/receiver.go

```diff
@@ -69,13 +70,10 @@ type ReceiverOption func(receiver *Receiver) error
 // for a queue or subscription.
 func ReceiverWithSubQueue(subQueue SubQueue) ReceiverOption {
 	return func(receiver *Receiver) error {
-		switch subQueue {
-		case SubQueueDeadLetter:
-		case SubQueueTransfer:
-		case "":
+		if subQueue == SubQueueDeadLetter || subQueue == SubQueueTransfer {
```

Comment:
I see this logic was duplicated elsewhere.  Consider moving it to some internal validation func or maybe a method on SubQueue depending on where it's needed.

Reply:
Funnily enough, I actually created a struct specifically for this kind of logic and never took advantage of this. Updated.